### ISSUE TITLE
valet share 301 redirect loop when used with valet secure on the most basic html site (BasicValetDriver)

### ISSUE THREAD

**Issue by vesper8** [laravel/valet#382]
Laravel Valet 2.0.4
Ngrok 2.2.4

Steps to reproduce the error:

cd ~/sites
mkdir test
cd test
echo "this is a test" > index.html

valet link test

visit test.dev in Chrome
see "this is a test"

valet driver
This site is served by [BasicValetDriver].

valet share

ngrok does its thing. Outputs a http and https url

visit either urls and they both work! alll good

now

valet secure

visit https://test.dev

works, no problem

valet share

visit either http or https ngrok urls

GET /                          301 Moved Permanently
GET /                          301 Moved Permanently
GET /                          301 Moved Permanently
GET /                          301 Moved Permanently
GET /                          301 Moved Permanently
GET /                          301 Moved Permanently
redirect loop

At first I was seeing this happen with Laravel sites.. but decided to see if I could reproduce the test with a basic html site as shown above, and sure enough the redirect loop occurs as soon as I use valet secure on my basic html site

**Comment by drbyte** [comment]
This is a known issue: valet share and valet secure don't work well together.

If you know a solution, a PR would be welcome.

One of a few related discussions: #148

**Comment by vesper8** [comment]
@drbyte Thank you for bringing this issue to my attention, I was quickly able to find the answer I was looking for in there.

Simply commenting out the initial server block in the generated nginx configuration file makes the whole thing work

```
server {
#    listen 80;
#    server_name subdomain.domain.dev;
#    return 301 https://$host$request_uri;
#}
```

After running
```
sudo brew services restart nginx
```

I am now able to run valet share and visit the https ngrok.io link without running into the redirect loop.

Furthermore I can still access the https:// version of my site so I didn't have to run valet unsecure as some other people suggested.

This is quite the easy fix.. the only cost is that the http no longer redirects to https on your local dev.. but that's no biggie at all.

Now I am trying to figure out how to pass arguments to valet share.

Running valet share -h doesn't offer any help other than list the ngrok help before crashing.

I tried valet share -subdomain=6ae11701 because I want to be able to request the same "subdomain" every time but that didn't work

**Comment by vesper8** [comment]
Found out how to get requested subdomains to work by using this excellent (rejected) :( PR https://github.com/laravel/valet/pull/81

Was so simple to add and works so well.. and it doesn't require a paid ngrok membership.

I really wish you would reconsider adding that PR @adamwathan  it's good to have options!

**Comment by bramus** [comment]
This can be closed now that https://github.com/laravel/valet/pull/630 has been merged _(and released in 2.1.3)_, no?
